src/util/url.ts

```diff
@@ -13,6 +15,32 @@ function getBaseUrl(url: URL): string {
   return baseUrl;
 }
 
+function getEndpointPath(
+  endpoint: ConfigurablePaths,
+  realmPath?: string,
+  customPaths?: CustomPathConfig,
+): string {
+  const realmUrlPath = getRealmUrlPath(realmPath);
+  const defaultPaths = {
+    authenticate: `json/${realmUrlPath}/authenticate`,
+    authorize: `oauth2/${realmUrlPath}/authorize`,
+    accessToken: `oauth2/${realmUrlPath}/access_token`,
+    endSession: `oauth2/${realmUrlPath}/connect/endSession`,
+    userInfo: `oauth2/${realmUrlPath}/userinfo`,
+    revoke: `oauth2/${realmUrlPath}/token/revoke`,
+    sessions: `json/${realmUrlPath}/sessions/`,
+  };
+  if (customPaths && customPaths[endpoint]) {
+    // TypeScript is not correctly reading the condition above
+    // It's thinking that customPaths[endpoint] may result in undefined
+    // eslint-disable-next-line
+    // @ts-ignore
+    return customPaths[endpoint];
```

Comment:
return (customPaths as CustomPathConfig)[endpoint]

might be cleaner?

Reply:
I tried this and a few other things, and unfortunately, it just doesn't like this block of code. It still refuses to properly understand the conditional, so I still need the comments above.